coderd/workspaces_test.go

```diff
@@ -1733,13 +1733,37 @@ func TestWorkspaceWatcher(t *testing.T) {
 	require.NoError(t, err)
 	wait("update workspace name", nil)
 
+	// Add a new version that will fail.
+	updatedVersion := coderdtest.CreateTemplateVersion(t, client, user.OrganizationID, &echo.Responses{
+		Parse:         echo.ParseComplete,
+		ProvisionPlan: echo.ProvisionComplete,
+		ProvisionApply: []*proto.Provision_Response{{
+			Type: &proto.Provision_Response_Complete{
+				Complete: &proto.Provision_Complete{
+					Error: "test error",
+				},
+			},
+		}},
+	}, func(req *codersdk.CreateTemplateVersionRequest) {
+		req.TemplateID = template.ID
+	})
+	coderdtest.AwaitTemplateVersionJob(t, client, updatedVersion.ID)
 	err = client.UpdateActiveTemplateVersion(ctx, template.ID, codersdk.UpdateActiveTemplateVersion{
-		ID: template.ActiveVersionID,
+		ID: updatedVersion.ID,
 	})
 	require.NoError(t, err)
 	wait("update active template version", nil)
 
-	cancel()
+	// Build with the new template; should end up with a failure state.
+	_ = coderdtest.CreateWorkspaceBuild(t, client, workspace, database.WorkspaceTransitionStart, func(req *codersdk.CreateWorkspaceBuildRequest) {
```

Comment:
For some reason creating this build fails the test, somewhere I think there is an assertion that the provisioner has no errors or something.

Reply:
Turns out it was the logger failing the test due to an error log when testing the failure state.

This implies to me that the logger's opinion is either that testing failures is an anti-pattern (which seems wrong) or that a provisioner job failing does not qualify to be logged as an error (could maybe change it to a warning).

For now I added a bool to disable this behavior rather than globally disable it or change the log level.